### ISSUE TITLE
when i try to deploy wfaster-whisper-server in offline server, docker run problem came up

### ISSUE THREAD

**Issue by james2v** [fedirz/faster-whisper-server#127]
(base) [root@app2 ~]# docker run --gpus all  -p 1080:8000 -v /app:/root/.cache/huggingface/ 784630b8bc0a

==========
== CUDA ==
==========

CUDA Version 12.2.2

Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.

error: Failed to prepare distributions
  Caused by: Failed to fetch wheel: faster-whisper-server @ file:///root/faster-whisper-server
  Caused by: Failed to install requirements from `build-system.requires` (resolve)
  Caused by: No solution found when resolving: hatchling
  Caused by: Could not connect, are you offline?
  Caused by: Request failed after 3 retries
  Caused by: error sending request for url (https://pypi.org/simple/hatchling/)
  Caused by: client error (Connect)
  Caused by: dns error: failed to lookup address information: Try again
  Caused by: failed to lookup address information: Try again

**Comment by james2v** [comment]
> I don't think you've included the full docker command in your description, as it doesn't have the image name. Are you using your own image or the one from Docker Hub?

784630b8bc0a is the image id of faster-whisper-server:latest-cuda, i am using the one form Doceker hub. i think the issus is that the when i  run the docker image, it automatically execute 
2024-10-21 09:22:43,704:INFO:httpx:_send_single_request:HTTP Request: GET https://api.gradio.app/gradio-messaging/en "HTTP/1.1 200 OK" , but in my offline server it fails. any chance i can find a image without gradio

**Comment by mindspawn** [comment]
Seeing the same issue as well. I like making my docker stacks use an isolated network to be sure no accidental data leaks. Supporting offline mode would be very useful.

**Comment by mindspawn** [comment]
@fedirz .. I think the root cause is because there are package dependencies in pyproject.toml that are not in the Dockerfile hence an attempt is made to download them which fails in offline deployments. e.g. hatchling, piper-phonemize etc.

**Comment by fedirz** [comment]
Yep, I've noticed this myself but didn't pay close attention to it, as I don't need to deploy the server in an offline environment. It is problematic, though, that the dependencies are being installed outside of the Docker build step. 

I've tried passing various combinations of flags to the `uv sync` and `uv run` commands but have been unable to resolve the issue. I believe that this problem is capture by [this](https://github.com/astral-sh/uv/issues/9191) `uv` issue

**Comment by fedirz** [comment]
@james2v @mindspawn The issue has been fixed in #206. I resolved it by removing the `uv run uvicorn` and instead calling `uvicorn` directly.

Thanks for creating the issue and using the project!
